webflow-hacks/form-hacks/42/42.ts

```diff
@@ -0,0 +1,27 @@
+// when dom is ready
+document.addEventListener('DOMContentLoaded', function () {
+  const CHECKBOX_SELECTOR = '[finsweet-hacks-element="checkbox"]';
+  const TOTAL_SELECTOR = '[finsweet-hacks-element="total-value"]';
+  const HIDDEN_INPUT_SELECTOR = '[finsweet-hacks-element="hidden-total"]';
+  const checkboxDivs = document.querySelectorAll(CHECKBOX_SELECTOR);
+  const totalValueDiv = document.querySelector<HTMLDivElement>(TOTAL_SELECTOR);
+  const hiddenTotalInput = document.querySelector<HTMLInputElement>(HIDDEN_INPUT_SELECTOR);
+  if (checkboxDivs.length === 0 || !totalValueDiv || !hiddenTotalInput) return;
+  let sum = 0;
+  // on each checkbox click
+  checkboxDivs.forEach((checkboxDiv) => {
+    checkboxDiv.addEventListener('click', function () {
+      // get the checkbox input field
+      const checkbox = checkboxDiv.previousElementSibling as HTMLInputElement;
+      if (!checkbox) return;
+      const amountToBeAdded = Number(checkbox.getAttribute('add-value'));
```

Comment:
You should check if the `amountToBeAdded` is invalid (`NaN`) as it could happen that the user writes invalid characters in the attribute.